### ISSUE TITLE
Helper to encode data for Quarto processing inside HTML table 

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-r#178]
See explanation at 
- https://github.com/haozhu233/kableExtra/issues/746#issuecomment-2095663392
related to 
- https://github.com/quarto-dev/quarto-cli/issues/555

Example: 
````R
as_quarto_markdown <- function(content) {
  sprintf('<span data-qmd="%s">%s</span>', content, content)
}
````

this could be used in table to make HTML with special attributes for Quarto to process and replace

- Should handle span and div. `as_quarto_markdown(..., block = TRUE)` ? 
- Should handle complex content by base64 encoding (using `xfun::base64_encode()`  and `data-qmd-base64` 
- Should probably have a markdown with special class to mark as it as content to not escape (using `I()` like for new R quarto inline ?)

**Comment by cderv** [comment]
>  do you plan to launch the new version to correct this issue. 

Hello @fgardavaud !

Can you help me understand better ? This is issue is about a feature request to quarto R package to add a helper for package author to use. (I added label to make this clearer). 

So adding this for next version won't solve any issues linked to kableExtra. They could already encode data the right way. 

Maybe I misunderstood though 🤷 

Anyhow, I'll add this in next version yes. Just need to make time on quarto R package instead of Quarto CLI.

**Comment by fgardavaud** [comment]
hello @cderv 
I potentially make a mistake on this post. I was facing with a problem with Mathjax. You could see this [page](https://stackoverflow.com/questions/79455156/why-is-mathjax-not-rendered-within-my-kableextra-table-in-a-quarto-document/79456551?noredirect=1#comment140129417_79456551) for the context and the solution :

My issue is only solve with html output and I hope there will be a solution for docx and pdf output.
My apologies for the confusion.

**Comment by cderv** [comment]
Yes this is related to the report in quarto-cli repo, and then ultimately this is **kableExtra**. 

This issue here is just derived from the other as an idea to offer a helper to encode date correctly to pass to quarto. 

I see the SO answer linked you here, as this is a workaround to format correctly the table content yourself, but this is not closing this issue that will solve your problem with kableExtra
